File: C++/2399. Check Distances Between Same Letters.cpp
class Solution {
public:
    bool checkDistances(string s, vector<int>& d) {
        vector<int> a(26, -1);
        for (int i = 0; i < s.size(); i++) {
            if (a[s[i] - 'a'] == -1)
                a[s[i] - 'a'] = i;
            else {
                a[s[i] - 'a'] = i - a[s[i] - 'a'] - 1;
                if (a[s[i] - 'a'] != d[s[i] - 'a'])
                    return false;
            }
        }
        return true;
    }
};

File: JAVA/2399. Check Distances Between Same Letters.java
class Solution {
    public boolean checkDistances(String s, int[] d) {
        int[] a=new int[26];
        for(int i=0;i<s.length();i++)
        a[i]--;
        for(int i=0;i<s.length();i++){
            int x=s.charAt(i)-'a';
            if(a[x]==-1)
            a[x]=i;
            else{
                a[x]=i-a[x]-1;
                if(a[x]!=d[x])
                return false;
            }
        }
        return true;
    }
}